### PR TITLE
ref: Electron forge config file

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
- Adds `forge.config.js` configuration file
- Updates ecma version
- Adds `package` script
- `package:<windows|mac|linux>` now build off the package script
- Consolidates the icons in `assets/icons`

### CODE REVIEW

File: forge.config.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        overwrite: true,
+      },
+      platforms: ["darwin"],
+    },
+    {
+      // Windows Distribution
+      name: "@electron-forge/maker-squirrel",
+      config: {
+        // TODO: Honeycomb icon (hexagon on docs)
+        // TODO: This will break when we merge with main
```
Comment (eldu) [inline forge.config.js]: Is this an issue for existing honeycomb users or is it just the icon so they'll get like a missing image?

**Review by eldu — COMMENTED** [review]
Passing through ! 🏃 

There are quite a few TODO statements here which look they they could be resolved by adding in the honeycomb icon png. Which could probably be done in this pr

I do like moving the forge config into it's own file though ! 

Also the check for the macOS build isn't going through

File: package.json [hunk 2]
```diff
@@ -89,9 +92,10 @@
     "build:firebase": "dotenv -e env/.env.firebase npm run build",
     "build:turk-prolific": "dotenv -e env/.env.turk-prolific npm run build",
     "rebuild": "electron-rebuild",
-    "package:windows": "react-scripts build && electron-forge make --arch x64 --targets @electron-forge/maker-squirrel",
-    "package:linux": "react-scripts build && electron-forge make --arch x64 --targets @electron-forge/maker-deb",
-    "package:mac": "react-scripts build && electron-forge make --arch x64 --targets @electron-forge/maker-dmg"
+    "package": "npm run build && electron-forge make",
+    "package:windows": "npm run package -- --arch x64 --targets @electron-forge/maker-squirrel",
+    "package:linux": "npm run package -- --arch x64 --targets @electron-forge/maker-deb",
+    "package:mac": "npm run package -- --arch x64 --targets @electron-forge/maker-dmg"
```
Comment (digicosmos86) [inline package.json]: Looks like this does not support ARM macs natively right? Is there an option that supports a `universal` build?

**Review by digicosmos86 — APPROVED** [review]
Looks great! I only have one question about native support for ARM macs.

File: forge.config.js (hunk 1 above)
Comment (RobertGemmaJr) [reply forge.config.js]: The URL is point to `main` and we're changing the path to that file so we'll have to update it when this PR gets into `main` (it's going into a different branch right now)

Comment (RobertGemmaJr) [reply forge.config.js]: Oh wait I can totally update the URL here for now, no one's going to be packaging based on this branch

**Comment by RobertGemmaJr** [comment]
> Also the check for the macOS build isn't going through

@digicosmos86 @eldu have you run into node gyp errors on Mac before? The build is failing on the `npm install` and I'm not sure why

File: package.json (hunk 2 above)
Comment (RobertGemmaJr) [reply package.json]: Looks like there is a universal value now, I'll swap it in!

**Comment by eldu** [comment]
> > Also the check for the macOS build isn't going through
> 
> @digicosmos86 @eldu have you run into node gyp errors on Mac before? The build is failing on the `npm install` and I'm not sure why

@RobertGemmaJr Perhaps upgrade node-gyp? It's at 10.0.1
https://github.com/nodejs/node-gyp

**Comment by eldu** [comment]
Oh node-gyp is not a direct dependency, nvm.

**Pull request merged**